src/algebra/order/monoid_lemmas.lean

```diff
@@ -224,19 +334,44 @@ alias mul_le_of_le_of_le_one ← mul_le_one'
 attribute [to_additive add_nonpos] mul_le_one'
 
 @[to_additive]
-lemma lt_mul_of_lt_of_one_le [covariant_class α α (*) (≤)]
-  {a b c : α} (hbc : b < c) (ha : 1 ≤ a) : b < c * a :=
-calc  b < c     : hbc
-    ... = c * 1 : (mul_one c).symm
-    ... ≤ c * a : mul_le_mul_left' ha c
+lemma mul_lt_of_le_of_lt_one [covariant_class α α (*) (<)]
+  {a b c : α} (hbc : b ≤ c) (ha : a < 1) : b * a < c :=
+calc  b * a < b * 1 : mul_lt_mul_left' ha b
+        ... = b     : mul_one b
+        ... ≤ c     : hbc
 
 @[to_additive]
 lemma mul_lt_of_lt_of_le_one [covariant_class α α (*) (≤)]
-  {a b c : α} (hbc : b < c) (ha : a ≤ 1)  : b * a < c :=
+  {a b c : α} (hbc : b < c) (ha : a ≤ 1) : b * a < c :=
 calc  b * a ≤ b * 1 : mul_le_mul_left' ha b
         ... = b     : mul_one b
         ... < c     : hbc
 
+@[to_additive]
+lemma mul_lt_of_lt_of_lt_one [covariant_class α α (*) (<)]
+  {a b c : α} (hbc : b < c) (ha : a < 1) : b * a < c :=
+calc  b * a < b * 1 : mul_lt_mul_left' ha b
+        ... = b     : mul_one b
+        ... < c     : hbc
+
+/-- Assumes left covariance. The lemma assuming right covariance is `right.mul_lt_one`. -/
+@[to_additive "Assumes left covariance. The lemma assuming right covariance is `right.add_neg`."]
+lemma left.mul_lt_one [covariant_class α α (*) (<)]
+  {a b : α} (ha : a < 1) (hb : b < 1) : a * b < 1 :=
+calc  a * b < a * 1 : mul_lt_mul_left' hb a
+        ... = a     : mul_one a
+        ... < 1     : ha
+
+-- Lemmas in the form of `b ≤ c → 1 ≤ a → b ≤ c * a`,
+-- which assume left covariance.
```

Comment:
Comments with this syntax will appear in the doc-gen output, which I think is a good idea:
```suggestion
/-! Lemmas in the form of `b ≤ c → 1 ≤ a → b ≤ c * a`,
which assume left covariance. -/
```